tests/Unit/Evolution/DgSubcell/Test_Reconstruction.cpp

```diff
@@ -0,0 +1,239 @@
+    CHECK_ITERABLE_CUSTOM_APPROX(reconstructed_datavector,
+                                 expected_nodal_coeffs, local_approx);
+
+    // Test reconstruction of a Variables
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>> cell_values_vars(
+        subcell_mesh.number_of_grid_points());
+    get(get<Tags::Scalar>(cell_values_vars)) =
+        TestHelpers::evolution::dg::subcell::cell_values(
+            std::min(dg_mesh.extents(0) - 2, 6_st), subcell_inertial_coords) *
+        projected_det_jac;
+    for (size_t d = 0; d < Dim; ++d) {
+      get<Tags::Vector<Dim>>(cell_values_vars).get(d) =
+          (d + 2.0) *
+          TestHelpers::evolution::dg::subcell::cell_values(
+              std::min(dg_mesh.extents(0) - 2, 6_st), subcell_inertial_coords) *
+          projected_det_jac;
+    }
+
+    const auto check_each_field_in_vars = [&dg_det_jac, &expected_nodal_coeffs,
+                                           &local_approx](
+                                              const auto& local_dg_vars) {
+      CHECK_ITERABLE_CUSTOM_APPROX(
+          DataVector(get(get<tmpl::front<typename std::decay_t<decltype(
+                             local_dg_vars)>::tags_list>>(local_dg_vars)) /
+                     dg_det_jac),
+          expected_nodal_coeffs, local_approx);
+      for (size_t d = 0; d < Dim; ++d) {
+        CHECK_ITERABLE_CUSTOM_APPROX(
+            DataVector(get<tmpl::back<typename std::decay_t<decltype(
+                           local_dg_vars)>::tags_list>>(local_dg_vars)
+                           .get(d) /
+                       dg_det_jac),
+            (d + 2.0) * expected_nodal_coeffs, local_approx);
+      }
+    };
+
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>>
+        reconstructed_vars = evolution::dg::subcell::fd::reconstruct(
+            cell_values_vars, dg_mesh, subcell_mesh.extents());
+    check_each_field_in_vars(reconstructed_vars);
+
+    // Check not_null with no prefix tags
+    reconstructed_vars.initialize(0);
+    check_each_field_in_vars(reconstructed_vars);
+
+    // Check with the prefix on the subcell vars
+    Variables<
+        tmpl::list<Tags::Prefix<Tags::Scalar>, Tags::Prefix<Tags::Vector<Dim>>>>
+        prefixed_cell_values_vars(subcell_mesh.number_of_grid_points());
+    prefixed_cell_values_vars = cell_values_vars;
+    evolution::dg::subcell::fd::reconstruct(make_not_null(&reconstructed_vars),
+                                            prefixed_cell_values_vars, dg_mesh,
+                                            subcell_mesh.extents());
+    check_each_field_in_vars(reconstructed_vars);
+
+    Variables<
+        tmpl::list<Tags::Prefix<Tags::Scalar>, Tags::Prefix<Tags::Vector<Dim>>>>
+        prefixed_reconstructed_vars{dg_mesh.number_of_grid_points()};
+    evolution::dg::subcell::fd::reconstruct(
+        make_not_null(&prefixed_reconstructed_vars), cell_values_vars, dg_mesh,
+        subcell_mesh.extents());
```

Comment:
Is this block supposed to have a comment like `// Check with the prefix on the DG vars` at the top and a check like `check_each_field_in_vars(prefixed_reconstructed_vars)` at the bottom???